sky/backends/backend_utils.py

```diff
@@ -813,9 +813,39 @@ def write_cluster_config(
 
     # Add kubernetes config fields from ~/.sky/config
     if isinstance(cloud, clouds.Kubernetes):
+        cluster_config_overrides = to_provision.cluster_config_overrides
+        if to_provision.network_tier is not None and to_provision.network_tier == resources_utils.NetworkTier.BEST:
+            k8s_env_vars = cast(Dict[str, Optional[str]],
+                                resources_vars['k8s_env_vars'])
+            if k8s_env_vars is not None:
+                context_name = k8s_env_vars.get(
+                    kubernetes.IN_CLUSTER_CONTEXT_NAME_ENV_VAR)
+                if context_name is not None and 'nebius' in context_name:
```

Comment:
context name can be easily changed by a user. We may need a more robust way to detect it.

Reply:
Can check if node has `nebius.com` label:
```
                    nebius.com/cuda_version=12.4.1-1
                    nebius.com/driverful=true
                    nebius.com/drivers-preset=cuda12
                    nebius.com/gpu=true
                    nebius.com/mlnx_openmpi_version=4.1.7a1-1.2310055
                    nebius.com/nccl_version=2.23.4-1-cuda12.4
                    nebius.com/node-group-id=mk8snodegroup-e00djm2ze8hp4fv78k
                    nebius.com/nvidia_driver_version=550.163.01-0ubuntu0.22.04.1
                    nebius.com/ofed_version=23.10-2.1.3.1
```